packages/esm-ward-app/src/ward-view/ward-view.component.tsx

```diff
@@ -11,22 +17,76 @@ const WardView = () => {
   const locationFromUrl = allLocations.find((l) => l.uuid === locationUuidFromUrl);
 
   if (locationUuidFromUrl && !locationFromUrl) {
-    showToast({
-      title: t('invalidLocationSpecified', 'Invalid location specified'),
-      kind: 'warning',
-      description: t('unknownLocationUuid', 'Unknown location uuid: {{locationUuidFromUrl}}', { locationUuidFromUrl }),
-    });
-
-    return <></>;
+    return (
+      <div className={styles.wardView}>
+        <div className={styles.wardViewMain}>
+          <InlineNotification
+            kind="error"
+            lowContrast={true}
+            title={t('invalidLocationSpecified', 'Invalid location specified')}
+            subtitle={t('unknownLocationUuid', 'Unknown location uuid: {{locationUuidFromUrl}}', {
+              locationUuidFromUrl,
+            })}
+          />
+        </div>
+      </div>
+    );
   }
 
-  const location = locationFromUrl ?? sessionLocation;
+  const location = (locationFromUrl ?? sessionLocation) as any as Location;
+  return <WardViewByLocation location={location} />;
+};
 
-  return (
-    <div>
-      <h1 id="ward-location">{location?.display}</h1>
-    </div>
-  );
+const WardViewByLocation = ({ location }: { location: Location }) => {
+  const { admissionLocation, isLoading, error } = useAdmissionLocation(location.uuid);
+  const { t } = useTranslation();
+
+  if (admissionLocation) {
+    // admissionLocation.bedLayouts can contain row+column positions with no bed,
+    // filter out layout positions with no real bed
+    const bedLayouts = admissionLocation.bedLayouts.filter((bl) => bl.bedId);
+
+    return (
+      <div className={styles.wardView}>
+        <div className={styles.wardViewHeader}>
+          <h4>{location?.display}</h4>
+        </div>
+        <div className={styles.wardViewMain}>
+          {bedLayouts.map((bedLayout, i) => {
+            const { patient } = bedLayout;
+            const bed = bedLayoutToBed(bedLayout);
+            return <WardBed key={bed.uuid} bed={bed} patients={patient ? [patient] : null} />;
+          })}
+          {bedLayouts.length == 0 && (
+            <InlineNotification
+              kind="warning"
+              lowContrast={true}
+              title={t('noBedsConfigured', 'No beds configured for this location')}
+            />
+          )}
+        </div>
+      </div>
+    );
+  } else if (isLoading) {
+    // TODO: add WardBedSkeleton
```

Comment:
Why is this a todo - is it pending a decision?  Can we implement and get rid of the hard-coded english text?